### DISCUSSION TITLE
how do i create new lua functions at runtime (from the java side)

### DISCUSSION THREAD

**Discussion by walksanatora** [cc-tweaked/CC-Tweaked#1395] [Q&A] [answered]
so lets say i have a type called eg: `interopModulePeripheral` that can only be gotten though `peripheral.call("casing_0","getFace","NORTH")`
how would i create the Table + functions at runtime

because calling
```java
    @LuaFunction
    public final void test(Object obj) {
        TISString.LOGGER.info(obj);
    }
 ```
 returns `null` when a function is passed

**Comment by walksanatora** [comment]
gonna see if `IArguments` has something

**Reply by walksanatora** [reply]
failed

**Comment by walksanatora** [comment, marked as answer]
Figured it out, you just have to implement `ILuaFunction` from `dan200.computercraft.api.lua`

**Comment by SquidDev** [comment]
I don't know quite what you're trying to do here, but if you're trying to return a table with string keys, you might be better off using [`IDynamicLuaObject`](https://tweaked.cc/mc-1.19.x/javadoc/dan200/computercraft/api/lua/IDynamicLuaObject.html), or just a Java class with `@LuaFunction`-annotated functions

**Reply by walksanatora** [reply]
... I could have done this
but what i ended up doing was making a "registry" that holds a Class (extending AbstractModle) and a Function<AbstractModule,Object>
which that Object gets passed straight to CC
